### PR TITLE
Project refactor, JSON and Tests

### PR DESCRIPTION
**Pull request by ajvazz**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,43 +1,41 @@
...
-Program provides information about videos/songs in a YouTube playlist. Idea behind this program is that some videos are sometimes deleted by uploaders, so this is a way to make sure you have saved at least song names (before they are deleted).
+Program provides information about videos/songs in a YouTube playlist. Idea behind this program is that some videos are sometimes deleted by uploaders, so this is a way to make sure you have saved at least song names before they are deleted.
 
 ## Starting the program
 
-#### Console
-If you don't want to clone this project, navigate to `jar` folder (or just  click [here](https://github.com/ajvazz/YouTube-Playlister/blob/master/YT_Playlister/jar/YT_Playlister.jar)) and download .jar file. Then, go to its download location, open terminal/console, and type:  
-`java -cp YT_Playlister.jar App`. Visit _Usage_ to see how to use the program.
+#### 1. Console
+If you don't want to clone this project, navigate to `target` folder (or just  click [here](https://github.com/ajvazz/YouTube-Playlister/blob/dragan/YT_Playlister/target/YouTubePlaylister.jar)) and download this .jar file. Then, go to its download location, open terminal/console, and type:
```
Comment (dragan-nikolic) [inline README.md]: jar file should not be stored in the git repo. Instead provide instructions how to build the jar file. Also provide info how to run the program from the command line (and provide at least one example).

Comment (dragan-nikolic) [reply README.md]: unable to run the program from command line. Got message: "no matches found: https://www.youtube.com/playlist?list=PL7DA3D097D6FDBC02"

File: README.md [hunk 2]
```diff
@@ -1,43 +1,41 @@
...
+`java -jar YouTubePlaylister.jar playlist_link path_to_output_file`.
 
-#### IntelliJ IDEA
-First, clone this repository with `git clone https://github.com/ajvazz/YouTube-Playlister.git`. Then, open IntelliJ IDEA and import this program (as Maven).  
+**For example**, if you'd like to save information about someone's [best 90s songs](https://www.youtube.com/playlist?list=PL7DA3D097D6FDBC02), you'd start the program like this:  
+`java -jar YouTubePlaylister.jar https://www.youtube.com/playlist?list=PL7DA3D097D6FDBC02 playlistOutput.json`.
+
+#### 2. IntelliJ IDEA
+First, clone this repository with `git clone https://github.com/ajvazz/YouTube-Playlister.git`. Then, open IntelliJ IDEA and import this program (as Maven). Since the program is run via _program arguments_, you would need to go to Edit Configurations in IntelliJ IDEA and under `Program arguments` just pass YouTube playlist's link and a file output, the same as you would in a console.
```
Comment (dragan-nikolic) [inline README.md]: unable to run program from IntelliJ. Got message "the working directory does not exist ..."

File: README.md [hunk 3]
```diff
@@ -1,43 +1,41 @@
...
 **Note**: there might be some IntelliJ-based problems with initial program start up. Feel free to contact me if you have any questions.
 
 ## Usage
+When you run this program as stated above (either via console or IntelliJ), an output file with playlist information will appear. It is located in path you provided when you started the program.
```
Comment (dragan-nikolic) [inline README.md]: playlist output file should be optional. Default value should be provided. At the end of execution program should print the location of the output file

Comment (dragan-nikolic) [reply README.md]: playlist ID is mandatory argument. If not provided, program should terminate and print appropriate message to the user.

File: YT_Playlister/src/main/java/Param.java [hunk 4]
```diff
@@ -1,31 +1,7 @@
-import io.restassured.response.Response;
-
-import java.util.ArrayList;
-import java.util.LinkedHashMap;
-
 public class Param {
```
Comment (dragan-nikolic) [inline YT_Playlister/src/main/java/Param.java]: what is the purpose of the Param.java class?

**Review by dragan-nikolic — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (ajvazz) [reply README.md]: > jar file should not be stored in the git repo.

I will remove it, and provide instructions on how to build it. 

> unable to run the program from command line. Got message: "no matches found: https://www.youtube.com/playlist?list=PL7DA3D097D6FDBC02"

Never seen that error. Did you start the jar file from dragan branch? (https://github.com/ajvazz/YouTube-Playlister/blob/dragan/YT_Playlister/target/YouTubePlaylister.jar). It works for me, I have started it like this:

java -jar YouTubePlaylister.jar https://www.youtube.com/playlist?list=PL7DA3D097D6FDBC02 output.json

File: README.md (hunk 2 above)
Comment (dragan-nikolic) [reply README.md]: this was issue with my run configuration, specified a file instead of a directory

**Review by dragan-nikolic — APPROVED** [review]

**Pull request merged**